virtualizarr/utils.py

```diff
@@ -0,0 +1,46 @@
+from typing import Optional, Union
+
+# TODO: importing fsspec and s3fs to get typing. Is there a better way incase these are optional deps?
+from s3fs.core import S3File
+from fsspec.implementations.local import LocalFileOpener
```

Comment:
You can use this try except pattern within `TYPE_CHECKING`

https://github.com/pydata/xarray/blob/aaa778cffb89baaece31882e03a7f4af0adfe798/xarray/core/types.py#L87